### PR TITLE
Adds rpc method for anvil to drop all pending transactions

### LINKED ISSUE #7249: Add a method to remove all pending txns from the txpool
### Component

Anvil

### Describe the feature you would like

`anvil_dropTranasction` already can drop transactions, iterating over with the response from `txpool_content` would be a quick endpoint.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mw2000**
## Motivation
Fixes #7249

## Solution
Added a new rpc method, and added the corresponding implementation in the mod.rs file for pool.

### CODE REVIEW

File: crates/anvil/src/eth/pool/mod.rs [hunk 1]
```diff
@@ -166,6 +166,16 @@ impl Pool {
...
     }
 
+    pub fn drop_all_transactions(&self) {
+        let pool = self.inner.write();
+        let mut transactions_iterator = pool.pending_transactions.transactions();
+        
+        while let Some(pool_transaction) = transactions_iterator.next() {
+            let tx_hash = pool_transaction.as_ref().hash();
+            self.drop_transaction(tx_hash);
+        }
```
Comment (mattsse) [inline crates/anvil/src/eth/pool/mod.rs:176]: hmm, I think we want a `fn clear` on both tx sets 

https://github.com/foundry-rs/foundry/blob/d8a162581aa2f674e581a57daf1aba24acc26206/crates/anvil/src/eth/pool/mod.rs#L202-L203

that clear all maps

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive

smol suggestion

**Comment by mw2000** [comment]
Now iterating over the ready transactions map and dropping them too.

**Review by mattsse — CHANGES_REQUESTED** [review]
style suggestion

File: crates/anvil/src/eth/pool/mod.rs [hunk 2]
```diff
@@ -166,6 +166,23 @@ impl Pool {
...
+        pending_transactions.for_each(|pool_transaction| {
+            let tx_hash = pool_transaction.as_ref().hash();
+            self.drop_transaction(tx_hash);
+        });
+
+        ready_transactions.for_each(|pool_transaction| {
+            let tx_hash = pool_transaction.as_ref().hash();
+            self.drop_transaction(tx_hash);
+        });
+    }
```
[the remark this answers is not in the document]
Comment (mw2000) [reply crates/anvil/src/eth/pool/mod.rs]: Makes sense, and this should be more performant too

**Comment by mw2000** [comment]
Moved to a `clear` function as requested @mattsse

**Review by mattsse — APPROVED** [review]
clean

**Pull request merged**